File: Mathematical Algorithms/Juggler Sequence/Juggler.java
public class Juggler {
    static void printJuggler(int n) {
        int a = n;
        // print the first term
        System.out.print(a + " ");
        // calculate terms until last term is not 1
        while (a != 1) {
            int b = 0;
            // Check if previous term is even or odd
            if (a % 2 == 0)
                // calculate next term
                b = (int) Math.floor(Math.sqrt(a));
            else
                // for odd previous term calculate next term
                b = (int) Math.floor(Math.sqrt(a) *  Math.sqrt(a) * Math.sqrt(a));
            System.out.print(b + " ");
            a = b;
        }
    }
}

File: Mathematical Algorithms/Juggler Sequence/Juggler.cpp
#include <cmath>
#include <iostream>
void printJuggler(long long n) {
    long long a = n;
    // print the first term
    std::cout << a << " ";
    // calculate terms until last term is not 1
    while (a != 1) {
        long long b = 0;
        // Check if previous term is even or odd
        if (a % 2 == 0)
            // calculate next term
            b = floor(sqrt(a));
        else
            // for odd previous term calculate next term
            b = floor(sqrt(a) * sqrt(a) * sqrt(a));
        std::cout << b << " ";
        a = b;
    }
}